src/js/models/document.js

```diff
@@ -172,7 +172,9 @@ define(function (require, exports, module) {
     Document.prototype.overlayVisibility = function (document) {
         var nextLayers = this.layers.overlayVisibility(document.layers);
 
-        return this.set("layers", nextLayers);
+        return this.set("layers", nextLayers)
+            .set("guidesVisible", this.guidesVisible)
+            .set("smartGuidesVisible", this.smartGuidesVisible);
```

Comment:
This is the other bug fix: retain guide/smartguide visibility from current state when navigating to a cached history state